### PR TITLE
Major breaking changes

### PR DESCRIPTION
**Pull request by ejball**
Ready for review.

### CODE REVIEW

**Comment by bgrainger** [comment]
Any thoughts on trying to make `StringSegment` API compatible with dotnet/corefx#20378?

**Comment by ejball** [comment]
Not a bad idea. But is their API stable yet?

File: src/Faithlife.Utility/EnumerableUtility.cs [hunk 1]
```diff
@@ -12,47 +12,13 @@ namespace Faithlife.Utility
...
 		/// <param name="first">The first sequence.</param>
 		/// <param name="second">The second sequence.</param>
 		/// <returns><c>True</c> if the sequences are equal.</returns>
 		public static bool AreEqual<T>(IEnumerable<T> first, IEnumerable<T> second)
-		{
-			return first != null ?
-				second != null && first.SequenceEqual(second) :
-				second == null;
-		}
+			=> first != null ? second != null && first.SequenceEqual(second) : second == null;
```
Comment (ddunkin) [inline src/Faithlife.Utility/EnumerableUtility.cs]: FWIW, I find this harder to read.

**Comment by ddunkin** [comment]
LGTM

**Pull request merged**